src/modules/advanced-custom-attributes/index.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { PanelAdvancedSettings } from '~stackable/components'
+import { i18n } from 'stackable'
+import { isUndefined, omit } from 'lodash'
+import striptags from 'striptags'
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks'
+import {
+	Fragment, useState, useEffect, render, unmountComponentAtNode,
+} from '@wordpress/element'
+import { __, sprintf } from '@wordpress/i18n'
+import { TextControl } from '@wordpress/components'
+
+const INVALID_HTML_ATTRIBUTES = [
+	'class',
+	'className',
+	'id',
+]
+
+const INVALID_BLOCK_ATTRIBUTES = [
+	'customAttributes',
+]
+
+const createAddSaveProps = ( extraProps, blockProps ) => {
+	if ( ! Array.isArray( blockProps.attributes.customAttributes ) || blockProps.attributes.customAttributes.length === 0 ) {
+		return extraProps
+	}
+
+	const customAttributes = {}
```

Comment:
Instead of defining a blank `customAttributes` then filling it up, it might be better to adjust the values directly.

```js
const customAttributes = Object.fromEntries( blockProps.attributes.customAttributes )
Object.keys( customAttributes ).forEach( key => {
    try {
        const value = customAttributes[ key ]
        // Replace the value if it's dynamic
        const dynamicAttributeMatch = value.match( /%([^\%]+)%/g )
        // ...
        customAttributes[ key ] = newValue
    } catch {}
} )
```

Reply:
I intentionally separated the custom attributes to properly omit invalid values from `INVALID_HTML_ATTRIBUTES`.